http.js

```diff
@@ -337,6 +337,12 @@ exports.request = function (request) {
             deferred.reject(error);
         });
 
+        if (request.socketTimeout) {
+            _request.setTimeout(request.socketTimeout, function() {
```

Comment:
This could be named simply "timeout" since it is an application layer timeout, not threaded through the TCP layer.